### PR TITLE
Model/audio search

### PR DESCRIPTION
**Pull request by titor-oopart**

### CODE REVIEW

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java [hunk 1]
```diff
@@ -21,54 +23,125 @@
...
-                    int initIndex = channelMode.indexOf(":");
-                    int endIndex = channelMode.length();
-                    int freeSpace = 2;
-                    channelMode = channelMode.substring(initIndex + freeSpace, endIndex);
-                    return channelMode;
+            while ((metadata = stdInput.readLine()) != null) {
+                audioChannel(metadata);
+                audioCodec(metadata);
+                mimeType(metadata);
+                sampleRAte(metadata);
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java:52]: Verify sample RAte name is correct

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java [hunk 2]
```diff
@@ -21,54 +23,125 @@
...
                 }
             }
         } catch (IOException e) {
-            System.out.println(e.getMessage());
+            e.printStackTrace();
             System.exit(-1);
+            System.out.println("No paso frame rate");
         }
-        return "there is not audio channel";
+        return "NO hay";
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java]: Verify return message

File: src/main/java/com/fundation/logic/common/MetadataCommonExtractor.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            while ((metadata = stdInput.readLine()) != null) {
+                String s = metadata;
+                list.add(metadata);
+                mimeType(metadata);
+                if ((metadata.contains("Read_All"))) {
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.exit(-1);
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/common/MetadataCommonExtractor.java:53]: Verify that this line not kill the application

File: src/main/java/com/fundation/logic/common/MetadataCommonExtractor.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                searchMimeType = "Video";
+            }
+            if (mimeType.contains("txt")) {
+                searchMimeType = "Text";
+            }
+            if (mimeType.contains("image")) {
+                searchMimeType = "Image";
+            }
+        }
+        System.out.println("No entro");
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/common/MetadataCommonExtractor.java]: Delete Sout

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java [hunk 5]
```diff
@@ -21,54 +23,125 @@
...
+                sampleRAte(metadata);
+                list.add(metadata);
+                if ((metadata.contains("Read_All"))) {
                 }
             }
         } catch (IOException e) {
-            System.out.println(e.getMessage());
+            e.printStackTrace();
             System.exit(-1);
+            System.out.println("No paso frame rate");
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java]: delete all system.out.println

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java (hunk 2 above)
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java]: change the message

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java (hunk 5 above)
Comment (aburgosu) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java]: Delete SOUT and System.exit()

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java [hunk 6]
```diff
@@ -21,54 +23,125 @@
...
+                mimeType(metadata);
+                sampleRAte(metadata);
+                list.add(metadata);
+                if ((metadata.contains("Read_All"))) {
                 }
             }
         } catch (IOException e) {
-            System.out.println(e.getMessage());
+            e.printStackTrace();
             System.exit(-1);
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java]: delete this line

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java [hunk 7]
```diff
@@ -21,54 +23,125 @@
...
+    public static String getAudioChannel(){
+        return searchChannelMode;
+    }
+
+    /**
+     * This method find audio codec .
+     */
+    public static void audioCodec(String audioCodec) {
+        if ((audioCodec.contains("MIME Type"))) {
+            if (audioCodec.contains("mpeg")){
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java]: space before {

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java [hunk 8]
```diff
@@ -21,54 +23,125 @@
...
+                searchAudioCodec = "MPEG-4 ALS";
+            }
+            if (audioCodec.contains("wav")){
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java]: space before

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java [hunk 9]
```diff
@@ -21,54 +23,125 @@
...
+                searchAudioCodec = "WAV";
+            }
+            if (audioCodec.contains("mp3")){
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java]: space before

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java [hunk 10]
```diff
@@ -21,54 +23,125 @@
...
-            e.printStackTrace();
-            System.exit(-1);
         }
-        return "There is not sample rate";
+    }
+
+    /**
+     * This method returns audrio codec to AudioSearch.
+     */
+    public static String searchAudioCodec(){
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java]: space before {

File: src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java [hunk 11]
```diff
@@ -21,54 +23,125 @@
...
+            if (mimeType.contains("audio")) {
+                searchMimeType = "audio";
+            }
+        }
+    }
+
+    /**
+     * This method returns mime type to AudioSearch.
+     */
+    public static String searchMimeType(){
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/common/MetadataAudioExtractor.java]: space before

File: src/main/java/com/fundation/logic/controller/Controller.java [hunk 12]
```diff
@@ -1,6 +1,6 @@
 /**
  * Copyright (c) 2019 Jalasoft.
- *
+ * <p>
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/controller/Controller.java]: delete <p>

File: src/main/java/com/fundation/logic/model/AudioSearch.java [hunk 13]
```diff
@@ -1,6 +1,6 @@
 /**
  * Copyright (c) 2019 Jalasoft.
- *
+ * <p>
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/model/AudioSearch.java:3]: delete <p>

File: src/main/java/com/fundation/logic/model/CommonSearch.java [hunk 14]
```diff
@@ -70,33 +72,44 @@ public List searchInPath(String path) {
...
-                if (evaluateString(fileName, criteriaFileName) &&
-                        evaluateString(fileExtension, criteriaExtension) &&
-                        evaluateHidden(fileHiddenStatus, criteriaHidden) &&
-                        evaluateReadOnly(fileCanWrite, criteriaReadOnly) &&
-                        evaluateSizeLimits(fileSize, sizeLowerLimit, sizeUpperLimit) &&
-                        evaluateDate(creationDate, creationDateLL, creationDateUL) &&
-                        evaluateDate(accessDate, accessDateLL, accessDateUL) &&
-                        evaluateDate(modificationDate, modificationDateLL, modificationDateUL) &&
-                        evaluateString(owner, criteriaOwner) && evaluateStringContains(mimeType, criteriaMimeType)){
-                    CustomizedFile matchingFile = new CustomizedFile(fileExtractor.getAbsolutePath(), fileName,
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/model/CommonSearch.java:95]: space before {

**Pull request merged**